SNIPS/snip-86.md

```diff
@@ -0,0 +1,601 @@
+
+- `domain` REQUIRED. An object containing information about the domain requesting the signing.
+
+  - `chainId` REQUIRED. The Starknet network identifier. Its value MUST be either "SN_SEPOLIA" or "SN_MAIN".
+  - `name` REQUIRED. The name of the application requesting the signing.
+  - `version` REQUIRED. The version of the application. Its value MUST be a string in the format x.y.z and cannot exceed 31 characters.
+  - `revision` REQUIRED. The Typed Data revision. Its value MUST be a string and MUST be "1" for this specification.
+
+- `message` REQUIRED. An object containing the core components of the SIWS message.
+
+  - `version` REQUIRED. The version of the SIWS message. Its value MUST be a string in the format x.y.z and cannot exceed 31 characters.
+  - `address` REQUIRED. The Starknet address performing the signing. Its value MUST be a hexadecimal string with 66 characters, including the '0x' prefix.
+  - `statement` REQUIRED. A human-readable ASCII assertion that the user will sign.
+  - `uri` REQUIRED. An RFC 3986 URI referring to the resource that is the subject of the signing.
+  - `nonce` REQUIRED. A random string used to prevent replay attacks, at least 8 alphanumeric characters and no more than 31 characters.
+  - `issuedAt` REQUIRED. The time when the message was generated. Its value MUST be an ISO 8601 datetime string.
+  - `validUntil` OPTIONAL. The time when the signed authentication message is no longer valid. Its value MUST be an ISO 8601 datetime string.
+  - `validAfter` OPTIONAL. The time when the signed authentication message will become valid. Its value MUST be an ISO 8601 datetime string.
+
+- `primaryType` REQUIRED. Specifies the primary type of the structured data. Its value MUST be "Message".
+
+- `types` REQUIRED. Defines the structure and types of the message and domain objects.
+
+  - `Message` REQUIRED. An array defining the structure of the message object.
+  - `StarknetDomain` REQUIRED. An array defining the structure of the domain object.
+
+#### Examples
+
+The following is an example of a basic SIWS message:
+
+```json
+{
+  "domain": {
+    "chainId": "SN_MAIN",
+    "name": "Example DApp",
+    "version": "1.0.0",
+    "revision": "1"
+  },
+  "message": {
+    "version": "1.0.0",
+    "address": "0x1234567890123456789012345678901234567890123456789012345678901234",
+    "statement": "Sign in with Starknet to Example DApp",
+    "uri": "https://example.com",
+    "nonce": "32891756",
+    "issuedAt": "2024-05-24T12:34:56Z",
+    "validUntil": "2024-05-24T13:34:56Z"
+  },
+  "primaryType": "Message",
+  "types": {
+    "Message": [
+      { "name": "version", "type": "string" },
+      { "name": "address", "type": "felt" },
+      { "name": "statement", "type": "string" },
+      { "name": "uri", "type": "string" },
+      { "name": "nonce", "type": "string" },
+      { "name": "issuedAt", "type": "string" },
+      { "name": "validUntil", "type": "string" }
```

Comment:
i'd rather remove this fields but if they are needed we should use `timestamp` from SNIP12 over strings